crates/sync/anvil/src/tracker.rs

```diff
@@ -265,18 +279,43 @@ pub async fn fetch_erc20_metadata(address: Address, client: &Provider<Http>) ->
     }
 }
 
-pub async fn fetch_erc721_metadata(
-    erc721_token_info: Erc721TokenInfo,
+pub async fn fetch_erc721_data(
+    erc721_token: Erc721Token,
     client: &Provider<Http>,
-) -> Erc721TokenMetadata {
-    let contract = IERC721::new(erc721_token_info.contract, Arc::new(client));
-    Erc721TokenMetadata {
+) -> Erc721TokenData {
```

Comment:
This needs to return `Result<Erc721TokenData>`, and we can't do `unwrap()`'s anywhere inside, since that'll completely break the tracker

you can refer to david's PR to see how this should be haandled (question mark operator), or you can ping me next week